eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -155,43 +155,45 @@ jobs:
           jdkVersionOption: '1.11'
           jdkArchitectureOption: 'x64'
           publishJUnitResults: false
-          goals: 'spotbugs:check checkstyle:checkstyle-aggregate'
+          goals: 'spotbugs:check checkstyle:checkstyle-aggregate javadoc:aggregate'
 
-      # Generate the Maven site report, including SpotBugs, JavaDoc, CheckStyle, Jacoco, etc only when we are not
-      # running as a PR check.
+      # We run a separate SpotBugs aggregate report step here to roll-up all the issues identified per-module
       - task: Maven@3
-        displayName: 'Generate Maven project site, including JavaDocs, SpotBugs, and CheckStyle reports'
+        displayName: 'Run SpotBugs aggregate analysis'
         inputs:
-          mavenPomFile: pom.client.xml
-          options: '$(DefaultOptions) -DskipTests -Dgpg.skip'
+          mavenPomFile: eng/spotbugs-aggregate-report/pom.xml
+          options: '--batch-mode -DskipTests -Dgpg.skip'
           mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
           javaHomeOption: 'JDKVersion'
           jdkVersionOption: '1.11'
           jdkArchitectureOption: 'x64'
           publishJUnitResults: false
-          goals: 'install site:site site:stage'
+          goals: 'install'
 
-      # We run a separate SpotBugs aggregate report step here to roll-up all the issues identified per-module in the
-      # previous step.
+      # Generate the Maven site report, including SpotBugs, JavaDoc, CheckStyle, Jacoco, etc only when we are not
+      # running as a PR check.
       - task: Maven@3
-        displayName: 'Run SpotBugs'
+        displayName: 'Generate Maven project site, including JavaDocs, SpotBugs, and CheckStyle reports'
+        condition: ne(variables['Build.Reason'], 'PullRequest')
         inputs:
-          mavenPomFile: eng/spotbugs-aggregate-report/pom.xml
-          options: '--batch-mode -DskipTests -Dgpg.skip'
+          mavenPomFile: pom.client.xml
+          options: '$(DefaultOptions) -DskipTests -Dgpg.skip'
           mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
           javaHomeOption: 'JDKVersion'
           jdkVersionOption: '1.11'
           jdkArchitectureOption: 'x64'
           publishJUnitResults: false
-          goals: 'install'
+          goals: 'install site:site site:stage'
```

Comment:
Note for reviewers: I switched the order of these two tasks, so that the more useful failures will be presented before the Maven site failures, which lack a lot of useful information.